programs/clearing_house/src/state/market.rs

```diff
@@ -227,6 +229,10 @@ pub struct PoolBalance {
 }
 
 impl SpotBalance for PoolBalance {
+    fn market_index(&self) -> u16 {
```

Comment:
so this is wrong for the revenue pool?

Reply:
cant think of a good way to do this given the zero copy of the explicit market_index caused so much pain...

Reply:
added a `BasePoolBalance` for revenue pool